src/System/Random/Internal.hs

```diff
@@ -806,6 +809,9 @@ uniformDoublePositive01M g = (+ d) <$> uniformDouble01M g
 instance UniformRange Float where
   uniformRM (l, h) g
     | l == h = return l
+    | isInfinite l && isInfinite h = return (0/0) -- NaN
```

Comment:
This was previously the case too, tested here: https://github.com/haskell/random/blob/11464aa406a3387dc61feab84149feb9f671a5fb/src/System/Random/Stateful.hs#L608-L611

But we need this guard now so we can distinguish "both endpoints are unequal infinities" (this case) from "one of the endpoints is an infinity" (the next two cases)..